src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseAnalysisController.java

```diff
@@ -444,30 +462,47 @@ public JsonResult updateCommonEntityInAnalysis(@PathVariable("analysisId") Long
+            throws PermissionDeniedException, NotExistException {
 
         IUser user = getUser(principal);
         T analysis = analysisService.getById(id);
+        List<UploadFileDTO> files = Stream.concat(uploadFilesLinks.getFiles().stream(), uploadFilesLinks.getLinks().stream()).collect(Collectors.toList());
+        return saveFiles(files, user, analysis);
+    }
 
-        JsonResult<List<AnalysisFileDTO>> result = new JsonResult<>(NO_ERROR);
+    private JsonResult<List<AnalysisFileDTO>> saveFiles(List<UploadFileDTO> files, IUser user, T analysis) {
 
+        Map<String, Object> validationErrors = new HashMap<>();
         List<AnalysisFileDTO> createdFiles = new ArrayList<>();
-
-        if (uploadFileDTO.getFile() != null) {
-            AnalysisFile createdFile = analysisService.saveFile(uploadFileDTO.getFile(), user, analysis, uploadFileDTO.getLabel(), uploadFileDTO.getExecutable(), null);
-            createdFiles.add(conversionService.convert(createdFile, AnalysisFileDTO.class));
-        } else {
-            if (StringUtils.hasText(uploadFileDTO.getLink())) {
-                AnalysisFile createdFile = analysisService.saveFile(uploadFileDTO.getLink(), user, analysis, uploadFileDTO.getLabel(), uploadFileDTO.getExecutable());
-                createdFiles.add(conversionService.convert(createdFile, AnalysisFileDTO.class));
-            } else {
-                result.setErrorCode(VALIDATION_ERROR.getCode());
+        List<String> errorFileMessages = new ArrayList<>();
+        JsonResult<List<AnalysisFileDTO>> result = new JsonResult<>(NO_ERROR);
+        files.forEach(uploadFile -> {
+            AnalysisFile createdFile = null;
+            try {
+                if (uploadFile.getFile() != null) {
+                    createdFile = analysisService.saveFile(uploadFile.getFile(), user, analysis, uploadFile.getLabel(), uploadFile.getExecutable(), null);
+                } else if (StringUtils.hasText(uploadFile.getLink())) {
+                    createdFile = analysisService.saveFile(uploadFile.getLink(), user, analysis, uploadFile.getLabel(), uploadFile.getExecutable());
+                } else {
+                    result.setErrorCode(VALIDATION_ERROR.getCode());
+                    return;
+                }
+            } catch (IOException e) {
+                LOGGER.error("Failed to save file", e);
+            } catch (AlreadyExistException e) {
+                LOGGER.error("Failed to save file", e);
+                result.setErrorCode(ALREADY_EXIST.getCode());
+                errorFileMessages.add(e.getMessage());
             }
+            createdFiles.add(conversionService.convert(createdFile, AnalysisFileDTO.class));
+        });
 
+        if (result.getErrorCode().equals(ALREADY_EXIST.getCode())) {
+            validationErrors.put("file", errorFileMessages.toArray(new String[0]));
+            result.setValidatorErrors(validationErrors);
```

Comment:
`ImmutableMap.of("file", errorFileMessages.toArray(new String[0]));`

and we do not need `validationErrors` var